From 52c521f03bf8e7df08dab3c011e01bb69daa3054 Mon Sep 17 00:00:00 2001
From: Olivier Moysan <olivier.moysan@foss.st.com>
Date: Tue, 29 Nov 2022 15:30:04 +0100
Subject: [PATCH 193/529] ARM: dts: stm32: increase ldo6 ramp delay on
 stm32mp15 eval boards

At initialization Wolfson wm8894 suspend/resume callbacks
are executed. On resume LDO6 regulator is re-enabled.
wm8994 Power On Reset must be in ready state to allow
accesses on control interface. Measurements on wm8994
show that POR is not in ready state before 170ms after
LDO6 activation. The LDO6 ramp-up delay is set to 300ms
to let enough time for wm8994 POR.

Signed-off-by: Olivier Moysan <olivier.moysan@foss.st.com>
Change-Id: I66bd776b1d3d3542eaa59c6b662489d18b09529f
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/278991
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
---
 arch/arm/boot/dts/stm32mp157a-ev1.dts | 10 ++++++++++
 arch/arm/boot/dts/stm32mp157c-ev1.dts | 10 ++++++++++
 arch/arm/boot/dts/stm32mp157d-ev1.dts | 10 ++++++++++
 arch/arm/boot/dts/stm32mp157f-ev1.dts | 10 ++++++++++
 4 files changed, 40 insertions(+)

--- a/arch/arm/boot/dts/stm32mp157a-ev1.dts
+++ b/arch/arm/boot/dts/stm32mp157a-ev1.dts
@@ -459,6 +459,16 @@
 	};
 };
 
+&i2c4 {
+	pmic: stpmic@33 {
+		regulators {
+			v1v8: ldo6 {
+				regulator-enable-ramp-delay = <300000>;
+			};
+		};
+	};
+};
+
 &i2c5 {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&i2c5_pins_a>;
--- a/arch/arm/boot/dts/stm32mp157c-ev1.dts
+++ b/arch/arm/boot/dts/stm32mp157c-ev1.dts
@@ -459,6 +459,16 @@
 	};
 };
 
+&i2c4 {
+	pmic: stpmic@33 {
+		regulators {
+			v1v8: ldo6 {
+				regulator-enable-ramp-delay = <300000>;
+			};
+		};
+	};
+};
+
 &i2c5 {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&i2c5_pins_a>;
--- a/arch/arm/boot/dts/stm32mp157d-ev1.dts
+++ b/arch/arm/boot/dts/stm32mp157d-ev1.dts
@@ -459,6 +459,16 @@
 	};
 };
 
+&i2c4 {
+	pmic: stpmic@33 {
+		regulators {
+			v1v8: ldo6 {
+				regulator-enable-ramp-delay = <300000>;
+			};
+		};
+	};
+};
+
 &i2c5 {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&i2c5_pins_a>;
--- a/arch/arm/boot/dts/stm32mp157f-ev1.dts
+++ b/arch/arm/boot/dts/stm32mp157f-ev1.dts
@@ -459,6 +459,16 @@
 	};
 };
 
+&i2c4 {
+	pmic: stpmic@33 {
+		regulators {
+			v1v8: ldo6 {
+				regulator-enable-ramp-delay = <300000>;
+			};
+		};
+	};
+};
+
 &i2c5 {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&i2c5_pins_a>;
